packages/dom/src/serialize-base64-images.js

```diff
@@ -0,0 +1,51 @@
+import { resourceFromText } from './utils';
+import { uid } from './prepare-dom';
+
+const signatures = {
+  JVBERi0: 'application/pdf',
+  R0lGODdh: 'image/gif',
+  R0lGODlh: 'image/gif',
+  iVBORw0KGgo: 'image/png',
+  '/9j/': 'image/jpg',
+  Qk02U: 'image/bmp'
+};
+
+function getBase64Substring(src) {
+  let base64Index = src.indexOf(';base64,');
+  if (base64Index === -1) return null;
+
+  base64Index += ';base64,'.length;
+  return src.substring(base64Index);
+}
+
+function detectMimeType(base64String) {
+  for (let s in signatures) {
+    if (base64String.indexOf(s) === 0) {
+      return signatures[s];
+    }
+  }
+  return null;
+}
+
+// Captures the current frame of videos and sets the poster image
+export function serializeBase64Images(dom, resources) {
```

Comment:
Can we generalize this for all data urls ? instead of specifically only doing for base 64 and images ?

Reply:
We also have svgs in the same fashion with utf8 data